File: backend/src/models/FakeTrendPost.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/database");

const User = require("./User");

const FakeTrendPost = sequelize.define("FakeTrendPost", {
	title: {
		type: DataTypes.STRING,
		allowNull: false,
	},
	trendName: {
		type: DataTypes.STRING,
		allowNull: false,
	},
	description: {
		type: DataTypes.TEXT,
		allowNull: false,
	},
	debunkExplanation: {
		type: DataTypes.TEXT,
		allowNull: false,
	},
	tiktokUrl: {
		type: DataTypes.STRING,
		allowNull: true,
	},
	tiktokVideoId: {
		type: DataTypes.STRING,
		allowNull: true,
	},
	videoUrl: {
		type: DataTypes.STRING,
		allowNull: true,
	},
	imageUrl: {
		type: DataTypes.STRING,
		allowNull: true,
	},
	skinConcernTag: {
		type: DataTypes.STRING,
		allowNull: true,
	},
	status: {
		type: DataTypes.ENUM("draft", "published"),
		defaultValue: "published",
	},
	dermatologistId: {
		type: DataTypes.INTEGER,
		allowNull: false,
	},
});

FakeTrendPost.belongsTo(User, {
	foreignKey: "dermatologistId",
	as: "dermatologist",
});

User.hasMany(FakeTrendPost, {
	foreignKey: "dermatologistId",
	as: "fakeTrendPosts",
});

module.exports = FakeTrendPost;

File: Cleari-FinalWork2026-Sophia-Soulaimane/Cleari-FinalWork2026-Sophia-Soulaimane/Features/AddDebunk/Models/FakeTrendPost.swift
//
//  FakeTrendPost.swift
//  Cleari-FinalWork2026-Sophia-Soulaimane
//
//  Created by Soulaimane Saadi on 22/05/2026.
//

import Foundation

struct FakeTrendFeedResponse: Codable {
    let posts: [FakeTrendPost]
}

struct FakeTrendPost: Codable, Identifiable {
    let id: Int
    let title: String
    let trendName: String
    let skinTypeTag: String?
    let skinConcernTag: String?
    let description: String
    let debunkExplanation: String
    let tiktokUrl: String?
    let tiktokVideoId: String?
    let videoUrl: String?
    let imageUrl: String?
    let status: String
    let dermatologistId: Int
    let createdAt: String
    let updatedAt: String
    let dermatologist: FakeTrendDermatologist?

    var likesCount: Int?
    var isLikedByCurrentUser: Bool?
    var commentsCount: Int?
    var savesCount: Int?
}

struct FakeTrendDermatologist: Codable {
    let id: Int
    let username: String
    let email: String
}

struct FakeTrendLikeResponse: Codable {
    let message: String
    let likesCount: Int
    let isLikedByCurrentUser: Bool
}
